### PR TITLE
[MM-45490 ] Adding compliance checks for threads

### PR DESCRIPTION
**Pull request by BenCookie95**
#### Summary
If compliance export is enabled we will hide top thread details for channels you are not a member of. Upon clicking the thread it will ask if you want to join the channel.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-45490

#### Related Pull Requests

- https://github.com/mattermost/mattermost-server/pull/20597

#### Screenshots

Compliance disabled:

Screen Shot 2022-07-07 at 6 10 38 AM

Compliance enabled:

Screen Shot 2022-07-07 at 6 11 01 AM

Modal on thread click:

Screen Shot 2022-07-07 at 6 11 12 AM

Join button clicked:

Screen Shot 2022-07-07 at 6 14 03 AM

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by BenCookie95** [comment]
@mkraft & @plant99 These are missing tests right now but I added you as reviewers anyway so you could get a first pass of it if you wanted

File: components/activity_and_insights/insights/join_channel_modal/join_channel_modal.test.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+                groups: {},
+                myGroups: [],
+            },
+            emojis: {
+                customEmoji: {},
+            },
+        },
+    };
+
+    test('should match snapshot', async () => {
```
Comment (plant99) [inline components/activity_and_insights/insights/join_channel_modal/join_channel_modal.test.tsx:126]: This means we should have new snapshot files as well?

**Comment by mkraft** [comment]
@BenCookie95 Github's UI seems to be glitching. I can see the successful POST request when @plant99 is added as a reviewer but he's not listed for me as a reviewer, even after a page refresh.

**Comment by mkraft** [comment]
@BenCookie95 It looks like the thread content is incorrectly being hidden for Professional license.

File: components/activity_and_insights/insights/top_threads/top_threads_table/top_threads_table.tsx [hunk 2]
```diff
@@ -87,11 +93,52 @@ const TopThreadsTable = (props: Props) => {
         props.closeModal();
     }, [props.closeModal]);
 
-    const openThread = (post: Post) => {
-        trackEvent('insights', 'open_thread_from_top_threads_modal');
-        dispatch(selectPostAndParentChannel(post));
```
Comment (mkraft) [inline components/activity_and_insights/insights/top_threads/top_threads_table/top_threads_table.tsx:92]: I think this is the only use of `selectPostAndParentChannel`, right? If so we could delete it.

**Review by mkraft — CHANGES_REQUESTED** [review]
Looks good. Just needs a fix for Professional license, which doesn't

**Comment by BenCookie95** [comment]
@mkraft Fixed for professional license now

**Comment by matthewbirtch** [comment]
@furqanmlk can we turn on the feature flag for the cloud test server?

Also @BenCookie95 will we need to turn on specific settings/licenses for the cloud test server to review this properly?

**Review by plant99 — APPROVED** [review]

**Review by matthewbirtch — APPROVED** [review]
Changes look good. I'll leave it to @furqanmlk to verify that this only is applied when compliance is enabled

**Comment by matthewbirtch** [comment]
> I've provided the following suggestions inline for the UI text elements for this new dialog:
> 
> * "This thread is from a channel you are not a member of and cannot be viewed unless you join the channel." --> "You must join the [channel-name] channel to see this thread."
> * "Join channel to view thread" --> "Join channel?"
> * "You are about to view a thread in the [channel-name] channel which you are not yet a member of. Do you want to join the channel in order to view the thread?" --> "You must join the [channel-name] channel to see this thread. Do you want to join [channel-name] now?"

@cwarnermm "You must join…` feels a bit strong IMHO. What do you think about "You'll need to join..."?

**Comment by cwarnermm** [comment]
Good feedback, @matthewbirtch! I chose "must" intentionally given the compliance context. However, "You'll need to join" works well too! Happy to go with that!

**Comment by BenCookie95** [comment]
@cwarnermm & @matthewbirtch I've made those updates, thanks for the help!

**Review by mkraft — APPROVED** [review]

File: components/activity_and_insights/insights/top_threads/top_threads_table/top_threads_table.tsx (hunk 2 above)
Comment (BenCookie95) [reply components/activity_and_insights/insights/top_threads/top_threads_table/top_threads_table.tsx:92]: Yeah we are using it further down on line 110, we just had to wrap it in a condition to determine if we wanted the modal or the RHS to open

**Pull request merged**